### ISSUE TITLE
Behaviour of CNP's when CIDR Group not existing

### ISSUE THREAD

**Issue by sass1997** [cilium/cilium#32735]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Not sure if it's a bug or the current normal behavior. Evendough it's a strange way to intepret cilium network policy with a cidr group which is currently not existing. My Policy does look like this (its part of a CCNP):
```
  - endpointSelector:
      matchLabels:
        "k8s:app.kubernetes.io/component": nginx-ingress
        "k8s:app.kubernetes.io/name": nginx-ingress
        "k8s:io.kubernetes.pod.namespace": ingress
    ingress:
      - fromCIDRSet:
          - cidrGroupRef: tcp-ingress
         toPorts:
          - ports:
              - port: "2222" # TCP Ingress Port
                protocol: TCP
```
If I apply this I get following logs in the agents
```
level=warning msg="Unable to translate all CIDR groups to CIDRs" cidrGroupRef="[tcp-ingress]" ciliumNetworkPolicyName=host-firewall error="cidr group \"tcp-ingress\" not found, skipping translation" k8sApiVersion= k8sNamespace= subsys=k8s-watcher
```
as well I can see that due to not able to translate all the CIDRS cilium is applying the rule with nodeselector: {}
```

\n&{EndpointSelector:{\"matchLabels\":{\"k8s:app.kubernetes.io/component\":\"nginx-ingress\",\"k8s:app.kubernetes.io/name\":\"nginx-ingress\",\"k8s:io.kubernetes.pod.namespace\":\"ingress\"}} NodeSelector:{} Ingress:[{IngressCommonRule:{FromEndpoints:] FromRequires:] FromCIDR: FromCIDRSet:] FromEntities:] aggregatedSelectors:]} ToPorts:] ICMPs:] Authentication:<nil>} {IngressCommonRule:{FromEndpoints:] FromRequires:] FromCIDR: FromCIDRSet:] FromEntities:] aggregatedSelectors:]} ToPorts:[{Ports:[{Port:2222 Protocol:TCP}] TerminatingTLS:<nil> OriginatingTLS:<nil> ServerNames:] Listener:<nil> Rules:<nil>}] ICMPs:] Authentication:<nil>}] IngressDeny:] Egress:] EgressDeny:] Labels:[:k8s:io.cilium.k8s.policy.derived-from=CiliumClusterwideNetworkPolicy k8s:io.cilium.k8s.policy.name=host-firewall k8s:io.cilium.k8s.policy.uid=b37af675-37d7-46c2-91e6-637c4f8bf66b] Description:}
```

Until I will add the specified CIDR Group everyone can access the port 2222 on the endpoint. My expection would be that some default value will be added to the cidr specification which will not open the port for everyone

### Cilium Version

1.51.1

### Kernel Version

5.15.0-105-generic

### Kubernetes Version

v1.26.15

### Cilium Users Document

- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by joestringer** [comment]
Thanks for the report. I agree that a policy referring to a group that doesn't exist shouldn't open up any traffic to be allowed; the "allow group X" should turn into "match nothing", not a  variation of "match everything".

**Comment by squeed** [comment]
I tested manually and confirmed this is still an issue, even after we refactored the policy ingestion code.

**Comment by squeed** [comment]
More fun:

**Allows port 22 to all**
```yaml
  egress:
    - toCIDRSet: [{cidrGroupRef: missing}]
      toPorts: { ... (port 22) ... }
```

**Blocks port 22**
```yaml
  egress:
    - toCIDRSet: []
      toPorts: { ... (port 22) ... }
```

**Blocks port 22**
```yaml
  egress:
    - toCIDR: []
      toPorts: { ... (port 22) ... }
```

**Comment by bimmlerd** [comment]
This is fixed for 1.16, but there's been discussions around whether we can backport this to 1.15 (which is what the report is actually about). The reason this is contentious is the fact that it constitutes a breaking change in CNP behaviour, which can lead to outages. Reopening for now.

**Comment by aanm** [comment]
@bimmlerd We can close this issue given the issue is fixed and it's going to be made available with v1.16.0.
